kstreams/backends/kafka.py

```diff
@@ -65,7 +65,7 @@ class Config:
         arbitrary_types_allowed = True
         use_enum_values = True
 
-    @root_validator
+    @root_validator(skip_on_failure=True)
```

Comment:
skip_on_failure must be true for pydantic v2. In this case it does not matter as this validator does not need to run if other validations already failed.

https://docs.pydantic.dev/latest/migration/#validator-and-root_validator-are-deprecated

Reply:
Shouldn't it be replaced by [model_validator](https://docs.pydantic.dev/latest/api/functional_validators/#pydantic.functional_validators.model_validator)?

Reply:
For pydantic v2, yes - but in this PR i am maintaining support for pydantic v1 so as not to be a breaking change